community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/SemanticState.scala

```diff
@@ -143,7 +143,9 @@ case class Scope(symbolTable: Map[String, Symbol], children: Seq[Scope]) extends
 object SemanticState {
   implicit object ScopeZipper extends TreeZipper[Scope]
 
-  val clean = SemanticState(Scope.empty.location, ASTAnnotationMap.empty, ASTAnnotationMap.empty)
+  val clean = cleanState(devNullLogger)
+
+  def cleanState(logger: NotificationLogger) = SemanticState(Scope.empty.location, ASTAnnotationMap.empty, ASTAnnotationMap.empty, logger)
```

Comment:
Can we add the devNullLogger as default and have a method called `withNotificationLogger` for setting the notification logger instead?